node/src/types/sync_leap.rs

```diff
@@ -128,7 +128,18 @@ impl SyncLeap {
     pub(crate) fn era_validator_weights(
         &self,
         fault_tolerance_fraction: Ratio<u64>,
+        protocol_config: &ProtocolConfig,
     ) -> impl Iterator<Item = EraValidatorWeights> + '_ {
+        // determine if the validator set has been updated in the
+        // current protocol version through an emergency upgrade
+        let validators_changed_in_current_protocol = protocol_config
+            .global_state_update
+            .as_ref()
+            .map_or(false, |global_state_update| {
+                global_state_update.validators.is_some()
```

Comment:
Is it enough to just check for `validators.is_some()`? If there are validators in the file, but the same as current ones, they aren't actually "changed".

Reply:
Yes, the check is enough to determine that no change was made to the validator set whatsoever. 
To compare the validators set from the file we need the validator set from global state when the upgrade was made. This info requires a read to global state and we want to avoid that with this optimization; there is already a mechanism for doing this based on the results of `global_states_for_sync_across_upgrade` which would include the era that was filtered out here.